app/controllers/users_controller.rb

```diff
@@ -1,6 +1,7 @@
 class UsersController < ApplicationController
   before_action :admin?, only: [:index, :show, :new, :create, :edit, :update, :destroy]
-  before_action :set_user, only: [:show, :edit, :update, :destroy]
+  before_action :set_user, only: [:show, :edit, :update, :destroy, :edit_password, :update_password]
+  before_action :current_user?, only: [:edit_password, :update_password]
```

Comment:
I tend to start to get worried when controllers state collection alot of `before_action` filters. It starts to make it difficult to folow the code path. "ok, what all method run before this controller action?", This gets even worse when we start having some before filters use `except:` and other use `only:`. If we pull  this into its own controller we could clean this chunk of code up some.